FGEGraphics/UISystem/UIElementText.cs

```diff
@@ -0,0 +1,179 @@
+        foreach (UIElementStyle style in Internal.ParentElement.ElementInternal.Styles)
+        {
+            if (!style.CanRenderText())
+            {
+                continue;
+            }
+            string styled = style.TextStyling(Internal.Content);
+            RenderableText text = style.TextFont.ParseFancyText(styled, style.TextBaseColor);
+            if (Internal.MaxWidth > 0)
+            {
+                // TODO: this does not work
+                text = FontSet.SplitAppropriately(text, Internal.MaxWidth);
+            }
+            Internal.RenderableContent[style] = text;
+        }
+    }
+
+    /// <summary>Gets or sets the raw text content.</summary>
+    public string Content
+    {
+        get => Internal.Content;
+        set
+        {
+            Internal.Content = value ?? (Required ? Null : null);
+            if (value is null)
+            {
+                if (!Empty)
+                {
+                    Empty = true;
+                    Internal.RenderableContent = null;
+                }
+            }
+            else
+            {
+                if (Empty)
+                {
+                    Empty = false;
+                    Internal.RenderableContent = [];
+                }
+                RefreshRenderables();
+            }
+        }
+    }
+
+    // TODO: check for 0 or negative?
+    /// <summary>Gets or sets the maximum total width of the text.</summary>
+    public int MaxWidth
+    {
+        get => Internal.MaxWidth;
+        set
+        {
+            Internal.MaxWidth = value;
+            RefreshRenderables();
+        }
+    }
+
+    /// <summary>
+    /// The <see cref="RenderableText"/> object corresponding to the parent element's current style.
+    /// Check <see cref="UIElementStyle.CanRenderText(UIElementText)"/> first.
+    /// </summary>
+    public RenderableText Renderable => Internal.RenderableContent?.GetValueOrDefault(Internal.ParentElement.ElementInternal.CurrentStyle, null);
+
+    /// <summary>The total width of the text.</summary>
+    public int Width => Renderable?.Width ?? 0;
+
+    /// <summary>The total height of the text.</summary>
+    public int Height => Renderable?.Lines.Length * Internal.ParentElement.ElementInternal.CurrentStyle.TextFont.FontDefault.Height ?? 0;
+
+    /// <summary>Returns the render position of the text given a starting position.</summary>
+    /// <param name="startX">The left-oriented anchor X value.</param>
+    /// <param name="startY">The top-oriented anchor Y value.</param>
+    public Location GetPosition(double startX, double startY)
+    {
+        double x = Math.Round(startX + HorizontalAlignment.SizeMultiplier() * -Width);
+        double y = Math.Round(startY + VerticalAlignment.SizeMultiplier() * -Height);
+        return new(x, y, 0);
+    }
+
+    /// <summary>
+    /// Returns <see cref="Renderable"/>.
+    /// Check <see cref="UIElementStyle.CanRenderText(UIElementText)"/> first.
```

Comment:
Having a special requirement to check before an _implicit operator_ gets called sounds like a recipe for surprise errors